var/spack/repos/builtin/packages/py-cython/package.py

```diff
@@ -37,42 +33,85 @@ class PyCython(PythonPackage):
-    depends_on("python@:3.11", type=("build", "link", "run"))
-
-    # https://github.com/cython/cython/commit/1cd24026e9cf6d63d539b359f8ba5155fd48ae21
-    # collections.Iterable was removed in Python 3.10
-    depends_on("python@:3.9", when="@:0.29.14", type=("build", "link", "run"))
-
-    # https://github.com/cython/cython/commit/430e2ca220c8fed49604daf578df98aadb33a87d
-    depends_on("python@:3.8", when="@:0.29.13", type=("build", "link", "run"))
-
-    depends_on("py-setuptools", type=("build", "run"))
+    with default_args(deprecated=True):
+        version(
+            "0.29.20", sha256="22d91af5fc2253f717a1b80b8bb45acb655f643611983fd6f782b9423f8171c7"
+        )
+        version(
+            "0.29.16", sha256="232755284f942cbb3b43a06cd85974ef3c970a021aef19b5243c03ee2b08fa05"
+        )
+        version(
+            "0.29.15", sha256="60d859e1efa5cc80436d58aecd3718ff2e74b987db0518376046adedba97ac30"
+        )
+        version(
+            "0.29.14", sha256="e4d6bb8703d0319eb04b7319b12ea41580df44fd84d83ccda13ea463c6801414"
+        )
+        version(
+            "0.29.13", sha256="c29d069a4a30f472482343c866f7486731ad638ef9af92bfe5fca9c7323d638e"
+        )
+        version(
+            "0.29.10", sha256="26229570d6787ff3caa932fe9d802960f51a89239b990d275ae845405ce43857"
+        )
+        version(
+            "0.29.7", sha256="55d081162191b7c11c7bfcb7c68e913827dfd5de6ecdbab1b99dab190586c1e8"
+        )
+        version(
+            "0.29.5", sha256="9d5290d749099a8e446422adfb0aa2142c711284800fb1eb70f595101e32cbf1"
+        )
+        version("0.29", sha256="94916d1ede67682638d3cc0feb10648ff14dc51fb7a7f147f4fedce78eaaea97")
+        version(
+            "0.28.6", sha256="68aa3c00ef1deccf4dd50f0201d47c268462978c12c42943bc33dc9dc816ac1b"
+        )
+        version(
+            "0.28.3", sha256="1aae6d6e9858888144cea147eb5e677830f45faaff3d305d77378c3cba55f526"
+        )
+        version(
+            "0.28.1", sha256="152ee5f345012ca3bb7cc71da2d3736ee20f52cd8476e4d49e5e25c5a4102b12"
+        )
+        version(
+            "0.25.2", sha256="f141d1f9c27a07b5a93f7dc5339472067e2d7140d1c5a9e20112a5665ca60306"
+        )
+        version(
+            "0.23.5", sha256="0ae5a5451a190e03ee36922c4189ca2c88d1df40a89b4f224bc842d388a0d1b6"
+        )
+        version(
+            "0.23.4", sha256="fec42fecee35d6cc02887f1eef4e4952c97402ed2800bfe41bbd9ed1a0730d8e"
+        )
+
+    depends_on("c", type="build")
+    depends_on("cxx", type="build")
+
+    # Based on PyPI wheel availability
+    with default_args(type=("build", "link", "run")):
+        depends_on("python@:3.13", when="@3.0.11:")
+        depends_on("python@:3.12", when="@0.29.35:3.0.10")
+        depends_on("python@:3.11", when="@0.29.29:0.29.34")
+        depends_on("python@:3.10", when="@0.29.25:0.29.28")
+        depends_on("python@:3.9", when="@0.29.21:0.29.24")
+        depends_on("python@:3.8", when="@0.29.14:0.29.20")
```

Comment:
I would drop all redundant lowerbounds, they just make it hard to read that the goal is to specify [forward compat bounds](https://spack.readthedocs.io/en/latest/packaging_guide.html#specifying-backward-and-forward-compatibility).
```suggestion
        depends_on("python@:3.13")  # or maybe even drop this line until 3.14 is released
        depends_on("python@:3.12", when="@:3.0.10")
        depends_on("python@:3.11", when="@:0.29.34")
        depends_on("python@:3.10", when="@:0.29.28")
        depends_on("python@:3.9", when="@:0.29.24")
        depends_on("python@:3.8", when="@:0.29.20")
```

Reply:
Would like to keep the 3.13 upper bound until we know which version will add 3.14 support. When 3.14 is added, we prob won't think to update this package too (unless CI catches an issue).